Creating `init_mocks` #91

bazkiaei opened the pull request:
The goal of this PR is to create a function which uses small functions of the code to provide the initials for the main functions of the code. We expect this function to be merged as `__init__` when the code starts to use classes.
This PR is related to issue #91

bazkiaei committed:
Creating `init_mocks` #91

bazkiaei commented on mocks/mocks.py:
```diff
@@ -23,6 +23,44 @@
+    # 'imager'. It is in progress in another PR so I prefer to wait for that.
+    huntsman = imagers['canon_sbig_dark']
+    psf_data = huntsman.psf.pixellated()
+    return config, huntsman, psf_data, cosmo, redshift
```
You can see that `config['total_mag']` is not included here. The reason is that to compute `config['total_mag']` someone needs the data from simulation while all other stuff that is computed in this function only needs `config`.

bazkiaei commented:
@AstroHuntsman/spiderlings 
Ready for review.

bazkiaei committed:
updating the PR

bazkiaei committed:
The imager (camera) which was added to the configuration yaml file in another PR, is used in the `init_mocks`

lspitler commented on mocks/mocks.py:
```diff
@@ -62,6 +62,42 @@ def crop_simulation_data(particle_pos,
+    -------
+    config : dict
+        A dictionary of configuration items.
+    huntsman : gunagala.imager.Imager
```
don't use huntsman here -- keep it generic so the docstring is relevant to any telescope we use

bazkiaei replied:
I changed it to `telescope`.

lspitler commented:
One small suggestion @bazkiaei . Without a sample jupyter notebook, I could not provide input on whether this has everything in it. I can do that when you're ready with the notebook review

bazkiaei committed:
updating the PR

bazkiaei committed:
jupyter notebood example updated

bazkiaei commented:
@lspitler 
Ready for review. The sample jupyter notebook is ready with respect to the latest changes in this PR.

bazkiaei committed:
Merge remote-tracking branch 'refs/remotes/upstream/develop' into mocks-init

lspitler commented on mocks/mocks.py:
```diff
@@ -68,6 +68,42 @@ def crop_simulation_data(particle_pos,
 
+def init_mocks(config):
+    """
+    Creates some initials for the code. This function will be converted to
```
```suggestion
    Initialises a mock instance. This function will be converted to
```

lspitler commented on mocks/mocks.py:
```diff
@@ -68,6 +68,42 @@ def crop_simulation_data(particle_pos,
+        Imager's psf data.
+    cosmo : astropy.cosmology.core.FlatLambdaCDM
+        The main cosmology that will be used by the code.
+    redshift : numpy.float
```
```suggestion
    redshift : float
```

lspitler commented on mocks/mocks.py:
```diff
@@ -68,6 +68,42 @@ def crop_simulation_data(particle_pos,
+                                                config['sim_pc_pixel'])
+    imagers = gunagala.imager.create_imagers()
+    telescope = imagers[config['imager']]
+    psf_data = telescope.psf.pixellated()
```
Suggest *not* returning psf_data as you then have two versions of the PSF in the notebook: `telescope.psf.pixellated()` and `psf_data`. Better to rely on gunagala class structure, so that someone can simply do:
```python
convolved = mocks.convolve_image_psf(noiseless,
                                    telescope.psf.pixellated(),
                                    convolution_boundary='extend')
```

bazkiaei replied:
@lspitler 
It absolutely makes sense what you suggested and I applied it.

lspitler commented:
@bazkiaei I can't test the notebook flow because it returns this error:
```python
OSError: File '/Users/lspitler/Documents/huntsman-mocks/examples/notebooks/sim_data/mw03.fits': format not understood or does not exist
```

It might be related to the fact that `snapshot_201` is not available in this branch, so I just changed it to another simulation file `mw03.fits` that doesn't appear to load.

Otherwise the notebook looks perfect - it loads everything in a few lines and provides good examples of what to do with your code. When you fix the issue above, I'll run the notebook and we can likely merge shortly after that.

I added some comments into the changed files. Please review in addition to fixing the simulation file problem.

bazkiaei commented:
> @bazkiaei I can't test the notebook flow because it returns this error:
> 
> ```python
> OSError: File '/Users/lspitler/Documents/huntsman-mocks/examples/notebooks/sim_data/mw03.fits': format not understood or does not exist
> ```
> 
> It might be related to the fact that `snapshot_201` is not available in this branch, so I just changed it to another simulation file `mw03.fits` that doesn't appear to load.
> 
> Otherwise the notebook looks perfect - it loads everything in a few lines and provides good examples of what to do with your code. When you fix the issue above, I'll run the notebook and we can likely merge shortly after that.
> 
> I added some comments into the changed files. Please review in addition to fixing the simulation file problem.

@lspitler 
That is because you do not have the right file on your computer in that folder. The input to the code should be a binary file at the moment, which is the output of the GENESIS, and not a fits file, that is why it does not work with the fits file which you have chosen. I am thinking about adding the ability of reading and processing fits files, but it is something to talk about. I have used one the snapshots of the GENESIS that you have provided. Here is the email that you sent me about that:
```
On 23 Nov 2018, at 3:34 pm, Lee Spitler <lee.spitler@mq.edu.au> wrote:

Hi Amir,

I managed to upload the rest of the data to here:
https://cloudstor.aarnet.edu.au/plus/s/hE9vHGNNFNG6Far 

I'm not sure what each set is, but it might become clearer once you open things.

Cheers,

Lee
```

Unfortunately, the link is not active anymore, but if you have those snapshots somewhere, it is the sixth of them, and I can provide that snapshot for you if you do not access them now.

bazkiaei committed:
lees suggestions applied

lspitler commented:
@bazkiaei let me know if you'd like input on this, otherwise its probably fine to merge.
